src/cart/src/cart/crt_iv.c

```diff
@@ -2485,6 +2494,9 @@ handle_ivupdate_response(const struct crt_cb_info *cb_info)
 	if (iv_info->uci_bulk_hdl != CRT_BULK_NULL)
 		crt_bulk_free(iv_info->uci_bulk_hdl);
 
+	if (iv_info->uci_sync_type.ivs_comp_cb)
+		iv_info->uci_sync_type.ivs_comp_cb(
+			iv_info->uci_sync_type.ivs_comp_cb_arg);
```

Comment:
Hmm i dont think calling it here is correct. First, if we are on intermediate node, this should not be called at all.
If we are not on intermediate node we issue sync completion only once sync has finished.

As it is now, this will execute sync cb right away if line 2480 ends up issuing asynchronous iv_sync.

The correct place to invoke ivs_comp_cb() would be in the completion callback of the iv_sync call

Reply:
ah, yes, you are right.